runtime/autoload/health/nvim.vim

```diff
@@ -0,0 +1,408 @@
+    let g:[host_skip_var] = old_skip
+  endif
+
+  if !empty(python_bin_name)
+    if exists('g:'.host_prog_var)
+      let python_bin = exepath(python_bin_name)
+    endif
+    let python_bin_name = fnamemodify(python_bin_name, ':t')
+  endif
+
+  if !empty(pythonx_errs)
+    call add(notes, pythonx_errs)
+  endif
+
+  if !empty(python_bin_name) && empty(python_bin) && empty(pythonx_errs)
+    if !exists('g:'.host_prog_var)
+      call health#report_warn(printf('"g:%s" is not set.  Searching for '
+            \ . '%s in the environment.', host_prog_var, python_bin_name))
+    endif
+
+    if !empty(pyenv)
+      if empty(pyenv_root)
+        call health#report_warn(
+              \ 'pyenv was found, but $PYENV_ROOT is not set.'
+              \ ['Did you follow the final install instructions?']
+              \ )
+      else
+        call health#report_ok(printf('pyenv found: "%s"', pyenv))
+      endif
+
+      let python_bin = s:trim(system(
+            \ printf('"%s" which %s 2>/dev/null', pyenv, python_bin_name)))
+
+      if empty(python_bin)
+        call health#report_warn(printf('pyenv couldn''t find %s.', python_bin_name))
+      endif
+    endif
+
+    if empty(python_bin)
+      let python_bin = exepath(python_bin_name)
+
+      if exists('$PATH')
+        for path in split($PATH, ':')
+          let path_bin = path.'/'.python_bin_name
+          if path_bin != python_bin && index(python_multiple, path_bin) == -1
+                \ && executable(path_bin)
+            call add(python_multiple, path_bin)
+          endif
+        endfor
+
+        if len(python_multiple)
+          " This is worth noting since the user may install something
+          " that changes $PATH, like homebrew.
+          call health#report_info(printf('There are multiple %s executables found.  '
+                \ . 'Set "g:%s" to avoid surprises.', python_bin_name, host_prog_var))
+        endif
+
+        if python_bin =~# '\<shims\>'
+          call health#report_warn(printf('"%s" appears to be a pyenv shim.', python_bin), [
+                      \ 'The "pyenv" executable is not in $PATH,',
+                      \ 'Your pyenv installation is broken. You should set '
+                      \ . '"g:'.host_prog_var.'" to avoid surprises.',
+                      \ ])
+        endif
+      endif
+    endif
+  endif
+
+  if !empty(python_bin)
+    if !empty(pyenv) && !exists('g:'.host_prog_var) && !empty(pyenv_root)
+          \ && resolve(python_bin) !~# '^'.pyenv_root.'/'
+      call health#report_warn('Your virtualenv is not set up optimally.', [
+            \ printf('Suggestion: Create a virtualenv specifically '
+            \ . 'for Neovim using pyenv and use "g:%s".  This will avoid '
+            \ . 'the need to install Neovim''s Python client in each '
+            \ . 'version/virtualenv.', host_prog_var)
+            \ ])
+    endif
+
+    if !empty(venv) && exists('g:'.host_prog_var)
```

Comment:
Yeah I'm pretty sure it should be `elseif !empty(venv) && !exists('g:'.host_prog_var)` since we're looking for a setup that's inferred.  The `if` statement above it should have `empty('venv') && ` as well.

The `pyenv` check is to see if the found python interpreter *is not* within `pyenv`'s managed paths.  `virtualenv` is suggested since I find it prudent to keep the base installations pristine.  I think I got confused because the message mentions `virtualenv`.  The message should be `pyenv is not setup optimally`.

I think it got janky because I was moving bits around.  If you don't mind, I can submit a PR to fix this portion.